getMultiStarts.m

```diff
@@ -222,40 +234,51 @@
         error_count = 0;
         
         % Test evaluation of objective function at starting point
+        % Only for multi-start local, since other optimizers use a 
+        % different initialization
         if (strcmp(options.localOptimizer, 'fmincon'))
             if (strcmp(options.localOptimizerOptions.Hessian, 'on'))
-                [J_0,~,~] = negLogPostWErrorCount(parameters.MS.par0(:,i)); % objectiveWrapWErrorCount(parameters.MS.par0(:,i),objective_function,options.obj_type,options.objOutNumber);
+                % We have to check Hessian AND gradient computation, since
+                % simulation can give different results if we're close to
+                % numerical failure
+                [J_0,~,~] = negLogPostWErrorCount(parameters.MS.par0(:,iMS));
```

Comment:
Unfortunately not right, the objective function gets called with two arguments.

According to the fmincon-documentation, the Hessian is not computed by the objective function, but passed as a function handle to fmincon in an option. So sometimes, the objective function gets called with two outputs, sometimes the Hessian function gets called.

I'll have a look, maybe there's a work around to have the objective function called with three putputs. But by default, fmincon doesn't do it when "options.Hessian" ="on".